sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2DataFrameSuite.scala

```diff
@@ -338,4 +341,166 @@ class DataSourceV2DataFrameSuite
+          new ColumnDefaultValue(
+            "('h' || 'r')",
+            new GeneralScalarExpression(
+              "CONCAT",
+              Array(
+                LiteralValue(UTF8String.fromString("h"), StringType),
+                LiteralValue(UTF8String.fromString("r"), StringType))),
+            LiteralValue(UTF8String.fromString("hr"), StringType)),
+          new ColumnDefaultValue(
+            "CAST(1 AS BOOLEAN)",
+            new V2Cast(LiteralValue(1, IntegerType), IntegerType, BooleanType),
+            LiteralValue(true, BooleanType))))
+
+      val df1 = Seq(1).toDF("id")
+      df1.writeTo(tableName).append()
+
+      sql(s"ALTER TABLE $tableName ALTER COLUMN dep SET DEFAULT ('i' || 't')")
+
+      val df2 = Seq(2).toDF("id")
+      df2.writeTo(tableName).append()
+
+      checkAnswer(
+        sql(s"SELECT * FROM $tableName"),
+        Seq(
+          Row(1, 123, "hr", true),
+          Row(2, 123, "it", true)))
+
+      val replaceExec = executeAndKeepPhysicalPlan[ReplaceTableExec] {
+        sql(
+          s"""
+             |REPLACE TABLE $tableName (
+             |  id INT,
+             |  salary INT DEFAULT (50 * 2),
+             |  dep STRING DEFAULT ('un' || 'known'),
+             |  active BOOLEAN DEFAULT CAST(0 AS BOOLEAN)
+             |) USING foo
+             |""".stripMargin)
+      }
+
+      checkDefaultValues(
+        replaceExec.columns,
+        Array(
+          null,
+          new ColumnDefaultValue(
+            "(50 * 2)",
+            new GeneralScalarExpression(
+              "*",
+              Array(LiteralValue(50, IntegerType), LiteralValue(2, IntegerType))),
+            LiteralValue(100, IntegerType)),
+          new ColumnDefaultValue(
+            "('un' || 'known')",
+            new GeneralScalarExpression(
+              "CONCAT",
+              Array(
+                LiteralValue(UTF8String.fromString("un"), StringType),
+                LiteralValue(UTF8String.fromString("known"), StringType))),
+            LiteralValue(UTF8String.fromString("unknown"), StringType)),
+          new ColumnDefaultValue(
+            "CAST(0 AS BOOLEAN)",
+            new V2Cast(LiteralValue(0, IntegerType), IntegerType, BooleanType),
+            LiteralValue(false, BooleanType))))
+
+      val df3 = Seq(1).toDF("id")
+      df3.writeTo(tableName).append()
+
+      checkAnswer(
+        sql(s"SELECT * FROM $tableName"),
+        Seq(Row(1, 100, "unknown", false)))
+    }
+  }
+
+  test("create/replace table with current like default values") {
+    val tableName = "testcat.ns1.ns2.tbl"
+    withTable(tableName) {
+      val createExec = executeAndKeepPhysicalPlan[CreateTableExec] {
+        sql(s"CREATE TABLE $tableName (id INT, cat STRING DEFAULT current_catalog()) USING foo")
+      }
+
+      checkDefaultValues(
+        createExec.columns,
+        Array(
+          null,
+          new ColumnDefaultValue(
+            "current_catalog()",
+            null, /* no V2 expression */
```

Comment:
we plan to support these current functions in DS v2, right?

Reply:
Correct, it is not supported yet.